### PR TITLE
homework04: Create pipes and directives

### PR DESCRIPTION
**Pull request by Zhydyk**

### CODE REVIEW

File: src/app/shared/directives/course-plate.directive.ts [hunk 1]
```diff
@@ -0,0 +1,28 @@
+import { Directive, ElementRef, HostListener, Input, OnInit, Renderer2 } from '@angular/core';
+
+@Directive({
+  selector: '[coursePlate]'
+})
+export class CoursePlateDirective implements OnInit{
```
Comment (SteveRusin) [inline src/app/shared/directives/course-plate.directive.ts:6]: create a separate module for directives and pipes. The same you did for compoenents

File: src/app/shared/directives/course-plate.directive.ts [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+  public changeBorder(): void {
+    const currentDate = +Date.now();
+    const creationDate = +new Date(this.date);
+    const milisecondsPerDay = 86400000;
+    const isFresh = creationDate < currentDate && creationDate >= (currentDate - 14 * milisecondsPerDay)
+    if (isFresh) {
+      this.render.addClass(this.element.nativeElement, 'fresh');
+    } 
+    if (creationDate > currentDate) {
```
Comment (SteveRusin) [inline src/app/shared/directives/course-plate.directive.ts]: use else, you don't need to check two conditions. If one is true the cond is obviously false

File: src/app/shared/pipes/order-by.pipe.ts [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+
+@Pipe({
+  name: 'orderBy'
+})
+export class OrderByPipe implements PipeTransform {
+
+  transform(course: Courses[]): Courses[] {
+    const sortByOrder = [...course];
+    return sortByOrder.sort((currentElement, previousElement) => {
+      const dataA = +new Date(currentElement.creationDate);
```
Comment (SteveRusin) [inline src/app/shared/pipes/order-by.pipe.ts]: move this callback to separate method or give it appropriate name, It's hard to get how you sort here.
`sortByOrder.sort(this.newCoursesFirst)` something like that

File: src/app/shared/directives/course-plate.directive.ts (hunk 2 above)
Comment (Zhydyk) [reply src/app/shared/directives/course-plate.directive.ts]: done

File: src/app/shared/pipes/order-by.pipe.ts (hunk 3 above)
Comment (Zhydyk) [reply src/app/shared/pipes/order-by.pipe.ts]: done

File: src/app/shared/directives/course-plate.directive.ts (hunk 1 above)
Comment (Zhydyk) [reply src/app/shared/directives/course-plate.directive.ts:6]: done

**Pull request merged**